File: Ruby/Euler016.rb
#!/usr/bin/env ruby

def main
  target = 1000

  x = 2 ** target
  sum = 0

  while x > 0
    x, mod = x.divmod(10)
    sum += mod
  end

  puts sum
end

if __FILE__ == $0
  main
end

File: Node/Euler016.js
#!/usr/bin/env node

function main() {
  const target = 1000n
  const big10 = 10n

  var x = 2n ** target
  var sum = 0n

  while (x > 0) {
    sum += x % big10
    x /= big10
  }

  console.log(String(sum))
}

if (require.main == module) {
  main();
}
